tools/osbuilder/dockerfiles/QAT/run.sh

```diff
@@ -0,0 +1,170 @@
+
+    if [ "${ROOTFS_OS}" == "debian" ] || [ "${ROOTFS_OS}" == "ubuntu" ]; then 
+        cd /proc
+    fi
+    /bin/echo -e "\n\e[1;42mDownload ${ROOTFS_OS} based rootfs\e[0m"
+    SECCOMP=no EXTRA_PKGS='kmod' ${katarepopath}/tools/osbuilder/rootfs-builder/rootfs.sh $ROOTFS_OS 
+}
+
+grab_qat_drivers()
+{
+    /bin/echo -e "\n\e[1;42mDownload and extract the drivers\e[0m" 
+    mkdir -p $QAT_SRC
+    cd $QAT_SRC
+    curl -L $QAT_DRIVER_URL | tar zx
+}
+
+build_qat_drivers()
+{
+    /bin/echo -e "\n\e[1;42mCompile driver modules\e[0m"
+    cd ${katarepopath}
+    linux_kernel_path=${katarepopath}/${LINUX_VER}
+    KERNEL_MAJOR_VERSION=$(awk '/^VERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+    KERNEL_PATHLEVEL=$(awk '/^PATCHLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+    KERNEL_SUBLEVEL=$(awk '/^SUBLEVEL =/{print $NF}' ${linux_kernel_path}/Makefile)
+    KERNEL_EXTRAVERSION=$(awk '/^EXTRAVERSION =/{print $NF}' ${linux_kernel_path}/Makefile)
+    KERNEL_ROOTFS_DIR=${KERNEL_MAJOR_VERSION}.${KERNEL_PATHLEVEL}.${KERNEL_SUBLEVEL}${KERNEL_EXTRAVERSION}
+    cd $QAT_SRC
+    KERNEL_SOURCE_ROOT=${linux_kernel_path} ./configure ${QAT_CONFIGURE_OPTIONS}
+    make all -j$(nproc) 
+}
+
+add_qat_to_rootfs()
+{
+    /bin/echo -e "\n\e[1;42mCopy driver modules to rootfs\e[0m"
+    cd $QAT_SRC
+    make INSTALL_MOD_PATH=${ROOTFS_DIR} qat-driver-install -j$(nproc)
+    cp $QAT_SRC/build/usdm_drv.ko ${ROOTFS_DIR}/lib/modules/${KERNEL_ROOTFS_DIR}/updates/drivers
+    depmod -a -b ${ROOTFS_DIR} ${KERNEL_ROOTFS_DIR}
+    cd ${katarepopath}/tools/osbuilder/image-builder
+    /bin/echo -e "\n\e[1;42mBuild rootfs image\e[0m"
+    ./image_builder.sh ${ROOTFS_DIR}
+}
+
+copy_outputs()
+{
+    /bin/echo -e "\n\e[1;42mCopy kernel and rootfs to the output directory and provide sample configuration files\e[0m"
+    mkdir -p ${OUTPUT_DIR} || true
+    cp ${linux_kernel_path}/arch/x86/boot/bzImage $OUTPUT_DIR/vmlinuz-${LINUX_VER}_qat
+    cp ${linux_kernel_path}/vmlinux $OUTPUT_DIR/vmlinux-${LINUX_VER}_qat
+    cp  ${katarepopath}/tools/osbuilder/image-builder/kata-containers.img $OUTPUT_DIR
+    mkdir -p ${OUTPUT_DIR}/configs || true
+    # Change extension from .conf.vm to just .conf and change the SSL section to 
+    # SHIM so it works with Kata containers 
+    for f in $QAT_SRC/quickassist/utilities/adf_ctl/conf_files/*.conf.vm; do
+        output_conf_file=$(basename -- "$f" .conf.vm).conf
+        cp -- "$f" "${OUTPUT_DIR}/configs/${output_conf_file}"
+        sed -i 's/\[SSL\]/\[SHIM\]/g' ${OUTPUT_DIR}/configs/${output_conf_file}
+    done
+}
+
+help() {
+cat << EOF
+Usage: $0 [-h] [options]
+   Description:
+        This script builds kernel and rootfs artifacts for Kata Containers,
+        configured and built to support QAT hardware.
+   Options:
+        -d,         Enable debug mode
+        -h,         Show this help
+EOF
+}
+
+main()
+{
+    local check_in_container=${OUTPUT_DIR:-}
+	if [ -z "${check_in_container}" ]; then
```

Comment:
Nit: inconsistent indent.

Reply:
Weird. In my editor it was correct.  I tapped out the spaces just to make sure it lines up.